### PR TITLE
Np 1204 update database service tests

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: src/test/java/no/unit/nva/database/DatabaseServiceImplTest.java [hunk 1]
```diff
@@ -39,37 +46,128 @@ public void getUserThrowsIllegalStateExceptionWhenItReceivesInvalidUserFromDatab
...
         Executable action = () -> service.getUser(someUser);
-        NotFoundException exception = assertThrows(NotFoundException.class, action);
-        assertThat(exception.getMessage(), containsString(DatabaseServiceImpl.USER_NOT_FOUND_MESSAGE));
+        IllegalStateException exception = assertThrows(IllegalStateException.class, action);
+
+        String expectedMessageContent = DatabaseServiceImpl.INVALID_ENTRY_IN_DATABASE_ERROR;
+        assertThat(exception.getMessage(), containsString(expectedMessageContent));
+    }
+
+    private DatabaseService mockServiceReceivingInvalidUserDbInstance() {
```
Comment (brinxmat) [inline src/test/java/no/unit/nva/database/DatabaseServiceImplTest.java]: Probably move the private methods to the bottom

File: src/test/java/no/unit/nva/database/DatabaseServiceTest.java [hunk 2]
```diff
@@ -42,8 +46,65 @@ public void databaseServiceHasAMethodForInsertingAUser()
...
+    public void databaseServiceReturnsNonEmptyRoleWhenRoleNameExistsInDatabase()
+        throws InvalidEntryInternalException, ConflictException, InvalidInputException, NotFoundException {
+        RoleDto insertedRole = createSampleRoleAndAddToDb(SOME_ROLE);
+        RoleDto savedRole = db.getRole(insertedRole);
+
+        assertThat(insertedRole, doesNotHaveNullFields());
+        assertThat(savedRole, is(equalTo(insertedRole)));
+    }
+
+    @DisplayName("getRole() throws NotFoundException when the role-name does exist in the database")
```
Comment (brinxmat) [inline src/test/java/no/unit/nva/database/DatabaseServiceTest.java]: does not

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by brinxmat — DISMISSED** [review]

File: src/test/java/no/unit/nva/database/DatabaseServiceImplTest.java (hunk 1 above)
Comment (axthosarouris) [reply src/test/java/no/unit/nva/database/DatabaseServiceImplTest.java]: fixed

File: src/test/java/no/unit/nva/database/DatabaseServiceTest.java (hunk 2 above)
Comment (axthosarouris) [reply src/test/java/no/unit/nva/database/DatabaseServiceTest.java]: fixed

**Review by brinxmat — APPROVED** [review]

**Pull request merged**